problem/templates/problem/problem_list.html

```diff
@@ -31,35 +31,55 @@
     </tr>
   </thead>
   <tbody>
-    {% for p in problem%}
+    {% for p in problem %}
       <tr>
         <td>
-          <span id='rejudge' class='glyphicon glyphicon-refresh' 
-          data-toggle="tooltip" data-placement="left" title="Rejudge"></span>&nbsp;{{p.pid}}</td>
-        <td><a href="{% url 'problem:problem.views.detail' p.pid %}">{{p.name}}</a>&nbsp;
-          <a href="{% url 'problem:problem.views.edit' p.pid %}">
-            <span id='edit' class='glyphicon glyphicon-pencil' 
-            data-toggle="tooltip" data-placement="right" title="Edit"></span>
-          </a>
+          {% if user == p.owner or user.is_admin %}
+            <span id='rejudge' class='glyphicon glyphicon-refresh' 
+            data-toggle="tooltip" data-placement="left" title="Rejudge"></span>
+          {% endif %}
+          &nbsp;{{p.pk}}
+        </td>
+
+        <td><a href="{% url 'problem:problem.views.detail' p.pk %}">{{p.pname}}</a>&nbsp;
+          {% if user == p.owner or user.is_admin %}
+            <a href="{% url 'problem:problem.views.edit' p.pk %}">
+              <span id='edit' class='glyphicon glyphicon-pencil' 
+              data-toggle="tooltip" data-placement="right" title="Edit"></span>
+            </a>
+          {% endif %}
         </td>
-        <script>
-        $("#rejudge").tooltip('hide');
-        $("#edit").tooltip('hide');
-        </script>
         <td>
-          <div class="progress">
-            <div class="progress-bar progress-bar-success" role="progressbar" aria-valuenow="{{p.pass}}" aria-valuemin="0" aria-valuemax="100" style="width: {{p.pass}}%;">
-              {{p.pass}}%
-            </div>
-            <div class="progress-bar progress-bar-danger" role="progressbar" aria-valuenow="{{p.not_pass}}" aria-valuemin="0" aria-valuemax="100" style="width: {{p.not_pass}}%;">
-              {{p.not_pass}}%
+          {% if p.no_submission %}
+            No submission
+          {% else %}
+            <div class="progress">
+              <div class="progress-bar progress-bar-success" style="width: {{p.pass_rate}}%;" 
+                data-toggle="tooltip" data-placement="top" title="{{p.pass_rate}}%">
```

Comment:
Use ``floatformat`` tag may be prettier.
https://docs.djangoproject.com/en/1.8/ref/templates/builtins/#floatformat